elm-src/NixOptions.elm

```diff
@@ -0,0 +1,50 @@
+module NixOptions exposing (..)
+import NixJSON exposing (Gson, gnull, gsonDecoder)
+import Json.Decode as Decode
+import Json.Decode.Pipeline exposing (decode, required, hardcoded, optional)
+
+type alias Option =
+    { declarations : List String
+    , default : Gson
+    , description : String
+    , example : Gson
+    , readOnly : Bool
+    , datatype : String
+    }
+
+
+type alias NixOptions =
+    List ( String, Option )
+
+option_sort_key : ( String, Option ) -> String
+option_sort_key ( name, option ) =
+    name
+
+option_filter : List String -> ( String, Option ) -> Bool
+option_filter terms ( name, option ) =
+    List.any (term_matches name) terms
+        || List.any (term_matches option.description) terms
+
+term_matches : String -> String -> Bool
+term_matches name term =
+    String.contains term (String.toLower name)
+
+
+filtered : NixOptions -> List String -> NixOptions
+filtered options terms =
+    List.filter (option_filter terms) options
+
+optionDecoder : Decode.Decoder Option
+optionDecoder =
+    decode Option
+        |> required "declarations" (Decode.list Decode.string)
+        |> optional "default" (Decode.lazy (\_ -> gsonDecoder)) gnull
```

Comment:
no need for `lazy` here

Reply:
I used `lazy` in both these cases because they're a bit expensive to decode, there are thousands of them, and a user may only ever need to look at 15 of them. Is this a sensible use of lazy?

Reply:
No, the decoded result is always fully evaluated. `lazy` applies to constructing the decoder, but makes no difference to actually executing the decoder.